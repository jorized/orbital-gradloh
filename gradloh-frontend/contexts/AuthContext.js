import React, { createContext, useState } from 'react';
import * as SecureStore from 'expo-secure-store';
const AuthContext = createContext(null);
const { Provider } = AuthContext;

const AuthProvider = ({ children }) => {


	const [authState, setAuthState] = useState({
		accessToken: null,
		refreshToken: null,
		authenticated: null
	});

	const logout = async () => {

		await SecureStore.deleteItemAsync('token');
		await SecureStore.deleteItemAsync('userdetails');
		await SecureStore.deleteItemAsync('selectedGradient');
		setAuthState({
			accessToken: null,
			refreshToken: null,
			authenticated: false,
			firstTimeUser: false
		});
	};

	const getAccessToken = () => {
		return authState.accessToken;
	};


	return (
		<Provider
			value={{
				authState,
				getAccessToken,
				setAuthState,
				logout,
			}}
		>
			{children}
		</Provider>
	);
};

export { AuthContext, AuthProvider };
