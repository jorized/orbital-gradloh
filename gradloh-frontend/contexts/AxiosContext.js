import React, { createContext, useContext, useEffect } from 'react';
import axios from 'axios';
import { AuthContext } from './AuthContext';
import { LoadingContext } from './LoadingContext'; // Import the Loading Context
import createAuthRefreshInterceptor from 'axios-auth-refresh';
import * as SecureStore from 'expo-secure-store';

const AxiosContext = createContext();
const authApiUrl = process.env.EXPO_PUBLIC_API_AUTH_URL;
const publicApiUrl = process.env.EXPO_PUBLIC_API_URL;
const { Provider } = AxiosContext;

const AxiosProvider = ({ children }) => {
  const authContext = useContext(AuthContext);
  const { setIsLoading } = useContext(LoadingContext); // Use the Loading Context
  const requestQueue = [];
  let isRefreshing = false;

  const processQueue = (error, token = null) => {
    requestQueue.forEach(promise => {
      if (error) {
        promise.reject(error);
      } else {
        promise.resolve(token);
      }
    });
    requestQueue.length = 0;
  };

      //Those who are not, use this route
    const publicAxios = axios.create({
      baseURL: authApiUrl
    });

  const authAxios = axios.create({
    baseURL: publicApiUrl,
    headers: {
      Authorization: `Bearer ${authContext.getAccessToken()}`,
    },
  });

  authAxios.interceptors.response.use(null, async (error) => {
    const originalRequest = error.config;
    if (error.response.status === 403 && !originalRequest._retry) {
      originalRequest._retry = true;
      if (!isRefreshing) {
        isRefreshing = true;
        setIsLoading(true); // Show loading indicator

        const data = {
          refreshToken: authContext.authState.refreshToken,
        };
        const options = {
          method: 'POST',
          data,
          url: `${authApiUrl}/refresh`,
        };

        try {
          const tokenRefreshResponse = await axios(options);
          isRefreshing = false;
          setIsLoading(false); // Hide loading indicator

          authContext.setAuthState({
            ...authContext.authState,
            accessToken: tokenRefreshResponse.data.accessToken,
          });

          await SecureStore.setItemAsync(
            'token',
            JSON.stringify({
              accessToken: tokenRefreshResponse.data.accessToken,
              refreshToken: authContext.authState.refreshToken,
            })
          );

          processQueue(null, tokenRefreshResponse.data.accessToken);

          originalRequest.headers.Authorization =
            'Bearer ' + tokenRefreshResponse.data.accessToken;

          return axios(originalRequest);
        } catch (e) {
          isRefreshing = false;
          setIsLoading(false); // Hide loading indicator
          processQueue(e, null);
          throw e;
        }
      }

      const retryOriginalRequest = new Promise((resolve, reject) => {
        requestQueue.push({ resolve, reject });
      });

      return retryOriginalRequest
        .then(token => {
          originalRequest.headers.Authorization = 'Bearer ' + token;
          return axios(originalRequest);
        })
        .catch(err => {
          return Promise.reject(err);
        });
    }
    return Promise.reject(error);
  });

  return (
    <Provider value={{ authAxios, publicAxios }}>
      {children}
    </Provider>
  );
};

export { AxiosContext, AxiosProvider };
