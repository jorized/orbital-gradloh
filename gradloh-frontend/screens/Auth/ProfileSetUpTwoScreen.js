import {
	View,
	Text,
	StyleSheet,
	TouchableOpacity,
	Platform,
	TouchableNativeFeedback,
	ActivityIndicator,
	LogBox
} from 'react-native';
import { useFonts } from 'expo-font';
import {
	Lexend_300Light,
	Lexend_400Regular,
	Lexend_600SemiBold,
	Lexend_700Bold
} from '@expo-google-fonts/lexend';
import React, { forwardRef, useContext, useState } from 'react';
import { useNavigation, useRoute } from '@react-navigation/native';
import DropdownWithErrorHandling from '../../components/Auth/DropdownWithErrorHandling';
import SelectableButton from '../../components/Auth/SelectableButton';
import { AxiosContext } from '../../contexts/AxiosContext';
import Toast from 'react-native-toast-message';
import { AuthContext } from '../../contexts/AuthContext';
import Faculties from '../../data/Faculties.json';
import PrimaryMajors from '../../data/PrimaryMajors.json';

// LogBox.ignoreLogs([
// 	'Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.'
// ]);

export default function ProfileSetUpTwoScreen() {
	const navigation = useNavigation();

	const [loading, setLoading] = useState(false);

	const authContext = useContext(AuthContext);
	const { publicAxios } = useContext(AxiosContext);

	const [facultyValue, setFacultyValue] = useState('');
	const [majorValue, setMajorValue] = useState('');
	const [isFacultyFocus, setIsFacultyFocus] = useState(false);
	const [isMajorFocus, setIsMajorFocus] = useState(false);
	const [isSelectedSecondMajor, setIsSelectedSecondMajor] = useState(false);
	const [isSelectedMinor, setIsSelectedMinor] = useState(false);
	const [facultyError, setFacultyError] = useState(false);
	const [majorError, setMajorError] = useState(false);

	const route = useRoute();
	const { nickname, email, password, enrolmentYear } = route.params || {};
	  
	//Loading in data
	const facultyData = Faculties;
	const majorData = PrimaryMajors;

	const filteredMajorData = facultyValue ? majorData[facultyValue] : [];

	const [fontsLoaded] = useFonts({
		Lexend_300Light,
		Lexend_400Regular,
		Lexend_600SemiBold,
		Lexend_700Bold
	});

	const handleSubmit = async () => {
		if (!facultyValue) {
			setFacultyError(true);
		}

		if (!majorValue) {
			setMajorError(true);
		} else {
			setFacultyError(false);
			setMajorError(false);
			// *Redirecting to different pages based on options*
			//If select second major, redirect to
			if (isSelectedSecondMajor) {
				navigation.push('ProfileSetUpThreeScreen', {
					nickname: nickname,
					email: email,
					password: password,
					enrolmentYear: enrolmentYear,
					homeFaculty: facultyValue,
					primaryMajor: majorValue,
					choseSecondMajor: isSelectedSecondMajor,
					choseMinor: isSelectedMinor
				});
			} else {
				if (isSelectedMinor) {
					//If only select minor
					navigation.push('ProfileSetUpFourScreen', {
						nickname: nickname,
						email: email,
						password: password,
						enrolmentYear: enrolmentYear,
						homeFaculty: facultyValue,
						primaryMajor: majorValue,
						choseSecondMajor: isSelectedSecondMajor,
						choseMinor: isSelectedMinor
					});
				} else {
					//If not go to onboarding
					setLoading(true);
					// Create the user account first
					try {
						const homeFaculty = facultyValue;
						const primaryMajor = majorValue;
						const secondaryMajor = '';
						const firstMinor = '';
						const secondMinor = '';
						const response = await publicAxios.post('/register', {
							nickname,
							email,
							password,
							enrolmentYear,
							primaryMajor,
							secondaryMajor,
							firstMinor,
							secondMinor,
							homeFaculty
						});
						setLoading(false);

						const { accessToken, refreshToken } = response.data;
						//Navigate to onboarding screen one
						authContext.setAuthState({
							accessToken,
							refreshToken
						});
						navigation.push('OnboardingScreen', {
							nickname,
							email,
							accessToken: accessToken,
							refreshToken: refreshToken
						});
					} catch (error) {
						console.log(error);
						setLoading(false);
						if (!error.response) {
							Toast.show({
								type: 'error',
								text1: 'Error',
								text2: 'Server is offline',
								visibilityTime: 5000,
								autoHide: true,
								position: 'bottom',
								bottomOffset: 40
							});
						} else {
							Toast.show({
								type: 'error',
								text1: 'Error',
								text2: error.response.data.message,
								visibilityTime: 5000,
								autoHide: true,
								position: 'bottom',
								bottomOffset: 40
							});
						}
					}
				}
			}
		}
	};

	if (!fontsLoaded) {
		return <Text>Loading...</Text>;
	}

	return (
		<View style={styles.container}>
			<View style={styles.form}>
				<Text style={styles.profileSetUpTitle}>Profile Setup</Text>
				<Text style={styles.facultyTitleText}>
					*Please select your faculty
				</Text>

				<DropdownWithErrorHandling
					data={facultyData}
					value={facultyValue}
					placeholder="Select faculty*"
					error={facultyError}
					setValue={setFacultyValue}
					setIsFocus={setIsFacultyFocus}
					setError={setFacultyError}
				/>

				{facultyValue && (
					<>
						<Text style={styles.majorTitleText}>
							*Please select your primary major
						</Text>
						<DropdownWithErrorHandling
							data={filteredMajorData}
							value={majorValue}
							placeholder="Select major*"
							error={majorError}
							setValue={setMajorValue}
							setIsFocus={setIsMajorFocus}
							setError={setMajorError}
							isSearch={true}
							searchPlaceholder="Search..."
						/>
					</>
				)}

				{majorValue && (
					<>
						<Text style={styles.optionalTitleText}>
							(Optional) Please select all that apply :
						</Text>
						<Text style={styles.optionalText}>
							I am currently taking a...
						</Text>
						<SelectableButton
							label="Second Major"
							isSelected={isSelectedSecondMajor}
							setIsSelected={setIsSelectedSecondMajor}
						/>
						<SelectableButton
							label="Minor"
							isSelected={isSelectedMinor}
							setIsSelected={setIsSelectedMinor}
						/>
					</>
				)}
			</View>
			{Platform.OS === 'android' ? (
				<TouchableNativeFeedback
					onPress={loading ? null : handleSubmit}
					background={TouchableNativeFeedback.Ripple('#fff', false)}
					disabled={loading}
				>
					<View
						style={[
							styles.loginPressable,
							loading && styles.disabledPressable
						]}
					>
						{loading ? (
							<ActivityIndicator size="small" color="#FFF" />
						) : (
							<Text style={styles.loginText}>Next</Text>
						)}
					</View>
				</TouchableNativeFeedback>
			) : (
				<TouchableOpacity
					style={[
						styles.loginPressable,
						loading && styles.disabledPressable
					]}
					onPress={loading ? null : handleSubmit}
					disabled={loading}
				>
					{loading ? (
						<ActivityIndicator size="small" color="#FFF" />
					) : (
						<Text style={styles.loginText}>Next</Text>
					)}
				</TouchableOpacity>
			)}
		</View>
	);
}

const styles = StyleSheet.create({
	container: {
		padding: 36,
		flex: 1,
		backgroundColor: 'white',
		justifyContent: 'space-between'
	},
	profileSetUpTitle: {
		fontFamily: 'Lexend_600SemiBold',
		color: '#2C2C2E',
		fontSize: 40,
		marginBottom: 24
	},
	facultyTitleText: {
		fontFamily: 'Lexend_300Light',
		fontSize: 16,
		marginBottom: 24
	},
	majorTitleText: {
		marginTop: 24,
		fontFamily: 'Lexend_300Light',
		fontSize: 16,
		marginBottom: 24
	},
	optionalTitleText: {
		marginTop: 24,
		fontFamily: 'Lexend_300Light',
		fontSize: 14
	},
	optionalText: {
		fontFamily: 'Lexend_300Light',
		fontSize: 14,
		marginBottom: 24
	},
	dropdownOne: {
		borderBottomWidth: 1,
		borderBottomColor: '#CBCBCB',
		marginBottom: 24
	},
	dropdownTwo: {
		borderBottomWidth: 1,
		borderBottomColor: '#CBCBCB',
		marginBottom: 24
	},
	dropdownPlaceholder: {
		color: '#CBCBCB'
	},
	loginPressable: {
		padding: 16,
		backgroundColor: '#EF7C00',
		borderRadius: 12
	},
	loginText: {
		textAlign: 'center',
		color: 'white',
		fontSize: 18,
		fontFamily: 'Lexend_400Regular'
	},
});
