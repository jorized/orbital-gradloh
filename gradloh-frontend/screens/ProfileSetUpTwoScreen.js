import {
	View,
	Text,
	StyleSheet,
	Image,
	TextInput,
	Pressable,
	TouchableOpacity,
	Platform,
	TouchableNativeFeedback,
	ActivityIndicator,
	LogBox
} from 'react-native';
import { useFonts } from 'expo-font';
import {
	Lexend_300Light,
	Lexend_400Regular,
	Lexend_600SemiBold,
	Lexend_700Bold
} from '@expo-google-fonts/lexend';
import { Dropdown } from 'react-native-element-dropdown';
import { forwardRef, useContext, useState } from 'react';
import { useNavigation, useRoute } from '@react-navigation/native';
import CheckBoxWithLabel from '../components/CheckBoxWithLabel';
import DropdownWithErrorHandling from '../components/DropdownWithErrorHandling';
import SelectableButton from '../components/SelectableButton';
import { AxiosContext } from '../contexts/AxiosContext';
import Toast from 'react-native-toast-message';
import { AuthContext } from '../contexts/AuthContext';

LogBox.ignoreLogs([
	'Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.'
]);

export default function ProfileSetUpTwoScreen() {
	const navigation = useNavigation();

	const [loading, setLoading] = useState(false);

	const authContext = useContext(AuthContext);
	const { publicAxios } = useContext(AxiosContext);

	const [facultyValue, setFacultyValue] = useState('');
	const [majorValue, setMajorValue] = useState('');
	const [isFacultyFocus, setIsFacultyFocus] = useState(false);
	const [isMajorFocus, setIsMajorFocus] = useState(false);
	const [isSelectedSecondMajor, setIsSelectedSecondMajor] = useState(false);
	const [isSelectedMinor, setIsSelectedMinor] = useState(false);
	const [facultyError, setFacultyError] = useState(false);
	const [majorError, setMajorError] = useState(false);

	const route = useRoute();
	const { nickname, email, password, enrolmentYear } = route.params || {};

	const toastConfig = {
		warning: ({ text1, text2, props }) => (
			<View style={[styles.toastContainer, styles.warningToast]}>
				<View style={styles.textContainer}>
					<Text style={styles.toastText1}>{text1}</Text>
					{text2 ? (
						<Text style={styles.toastText2}>{text2}</Text>
					) : null}
				</View>
				<TouchableOpacity
					onPress={() => Toast.hide()}
					style={styles.closeButton}
				>
					<Text style={styles.closeButtonText}>✕</Text>
				</TouchableOpacity>
			</View>
		),
		success: ({ text1, text2, props }) => (
			<View style={[styles.toastContainer, styles.successToast]}>
				<View style={styles.textContainer}>
					<Text style={styles.toastText1}>{text1}</Text>
					{text2 ? (
						<Text style={styles.toastText2}>{text2}</Text>
					) : null}
				</View>
				<TouchableOpacity
					onPress={() => Toast.hide()}
					style={styles.closeButton}
				>
					<Text style={styles.closeButtonText}>✕</Text>
				</TouchableOpacity>
			</View>
		)
	};

	const CustomToast = forwardRef((props, ref) => (
		<Toast ref={ref} config={toastConfig} />
	));

	const facultyData = [
		{ label: 'Arts and Social Sciences (FASS)', value: 'FASS' },
		{ label: 'Business (SOB)', value: 'SOB' },
		{ label: 'Computing (SOC)', value: 'SOC' },
		{ label: 'Design and Engineering (CDE)', value: 'CDE' },
		{ label: 'Science (FOS)', value: 'FOS' }
	];

	const majorData = {
		FASS: [
			{ label: 'Anthropology', value: 'Anthropology' },
			{ label: 'Chinese Studies', value: 'Chinese Studies' },
			{
				label: 'Communications and New Media',
				value: 'Communications and New Media'
			},
			{ label: 'Economics', value: 'Economics' }
			// Add other majors for FASS
		],
		SOB: [
			{
				label: 'Business Administration',
				value: 'Business Administration'
			},
			{ label: 'Accountancy', value: 'Accountancy' }
			// Add other majors for SOB
		],
		SOC: [
			{ label: 'Business Analytics', value: 'Business Analytics' },
			{ label: 'Computer Engineering', value: 'Computer Engineering' },
			{ label: 'Computer Science', value: 'Computer Science' },
			{ label: 'Information Security', value: 'Information Security' },
			{ label: 'Information Systems', value: 'Information Systems' }
			// Add other majors for SOC
		],
		CDE: [
			{ label: 'Architecture', value: 'Architecture' },
			{ label: 'Industrial Design', value: 'Industrial Design' },
			{ label: 'Real Estate', value: 'Real Estate' },
			{ label: 'Biomedical Engineering', value: 'Biomedical Engineering' }
			// Add other majors for CDE
		],
		FOS: [
			{ label: 'Applied Mathematics', value: 'Applied Mathematics' },
			{ label: 'Chemistry', value: 'Chemistry' },
			{ label: 'Computational Biology', value: 'Computational Biology' },
			{
				label: 'Data Science and Analytics',
				value: 'Data Science and Analytics'
			}
			// Add other majors for FOS
		]
	};
	const filteredMajorData = facultyValue ? majorData[facultyValue] : [];

	const [fontsLoaded] = useFonts({
		Lexend_300Light,
		Lexend_400Regular,
		Lexend_600SemiBold,
		Lexend_700Bold
	});

	const handleSubmit = async () => {
		if (!facultyValue) {
			setFacultyError(true);
		}

		if (!majorValue) {
			setMajorError(true);
		} else {
			setFacultyError(false);
			setMajorError(false);
			// *Redirecting to different pages based on options*
			//If select second major, redirect to
			if (isSelectedSecondMajor) {
				navigation.push('ProfileSetUpThreeScreen', {
					nickname: nickname,
					email: email,
					password: password,
					enrolmentYear: enrolmentYear,
					homeFaculty: facultyValue,
					primaryMajor: majorValue,
					choseSecondMajor: isSelectedSecondMajor,
					choseMinor: isSelectedMinor
				});
			} else {
				if (isSelectedMinor) {
					//If only select minor
					navigation.push('ProfileSetUpFourScreen', {
						nickname: nickname,
						email: email,
						password: password,
						enrolmentYear: enrolmentYear,
						homeFaculty: facultyValue,
						primaryMajor: majorValue,
						choseSecondMajor: isSelectedSecondMajor,
						choseMinor: isSelectedMinor
					});
				} else {
					//If not go to onboarding
					setLoading(true);
					// Create the user account first
					try {
						const homeFaculty = facultyValue;
						const primaryMajor = majorValue;
						const secondaryMajor = '';
						const firstMinor = '';
						const secondMinor = '';
						const response = await publicAxios.post('/register', {
							nickname,
							email,
							password,
							enrolmentYear,
							primaryMajor,
							secondaryMajor,
							firstMinor,
							secondMinor,
							homeFaculty
						});
						setLoading(false);

						const { accessToken, refreshToken } = response.data;
						//Navigate to onboarding screen one
						authContext.setAuthState({
							accessToken,
							refreshToken
						});
						navigation.push('OnboardingScreen', {
							email,
							accessToken: accessToken,
							refreshToken: refreshToken
						});
					} catch (error) {
						console.log(error);
						setLoading(false);
						if (!error.response) {
							Toast.show({
								type: 'warning',
								text1: 'Error',
								text2: 'Server is offline',
								visibilityTime: 5000,
								autoHide: true,
								position: 'bottom',
								bottomOffset: 40
							});
						} else {
							Toast.show({
								type: 'warning',
								text1: 'Error',
								text2: error.response.data.message,
								visibilityTime: 5000,
								autoHide: true,
								position: 'bottom',
								bottomOffset: 40
							});
						}
					}
				}
			}
		}
	};

	if (!fontsLoaded) {
		return <Text>Loading...</Text>;
	}

	return (
		<View style={styles.container}>
			<View style={styles.form}>
				<Text style={styles.profileSetUpTitle}>Profile Setup</Text>
				<Text style={styles.facultyTitleText}>
					*Please select your faculty
				</Text>

				<DropdownWithErrorHandling
					data={facultyData}
					value={facultyValue}
					placeholder="Select faculty*"
					error={facultyError}
					setValue={setFacultyValue}
					setIsFocus={setIsFacultyFocus}
					setError={setFacultyError}
				/>

				{facultyValue && (
					<>
						<Text style={styles.majorTitleText}>
							*Please select your primary major
						</Text>
						<DropdownWithErrorHandling
							data={filteredMajorData}
							value={majorValue}
							placeholder="Select major*"
							error={majorError}
							setValue={setMajorValue}
							setIsFocus={setIsMajorFocus}
							setError={setMajorError}
							isSearch={true}
							searchPlaceholder="Search..."
						/>
					</>
				)}

				{majorValue && (
					<>
						<Text style={styles.optionalTitleText}>
							(Optional) Please select all that apply :
						</Text>
						<Text style={styles.optionalText}>
							I am currently taking a...
						</Text>
						<SelectableButton
							label="Second Major"
							isSelected={isSelectedSecondMajor}
							setIsSelected={setIsSelectedSecondMajor}
						/>
						{/* <SelectableButton
                            label="Minor" 
                            isSelected={isSelectedMinor} 
                            setIsSelected={setIsSelectedMinor} 
                        /> */}
					</>
				)}
			</View>
			{Platform.OS === 'android' ? (
				<TouchableNativeFeedback
					onPress={loading ? null : handleSubmit}
					background={TouchableNativeFeedback.Ripple('#fff', false)}
					disabled={loading}
				>
					<View
						style={[
							styles.loginPressable,
							loading && styles.disabledPressable
						]}
					>
						{loading ? (
							<ActivityIndicator size="small" color="#FFF" />
						) : (
							<Text style={styles.loginText}>Next</Text>
						)}
					</View>
				</TouchableNativeFeedback>
			) : (
				<TouchableOpacity
					style={[
						styles.loginPressable,
						loading && styles.disabledPressable
					]}
					onPress={loading ? null : handleSubmit}
					disabled={loading}
				>
					{loading ? (
						<ActivityIndicator size="small" color="#FFF" />
					) : (
						<Text style={styles.loginText}>Next</Text>
					)}
				</TouchableOpacity>
			)}
			<CustomToast />
		</View>
	);
}

const styles = StyleSheet.create({
	container: {
		padding: 36,
		flex: 1,
		backgroundColor: 'white',
		justifyContent: 'space-between'
	},
	profileSetUpTitle: {
		fontFamily: 'Lexend_600SemiBold',
		color: '#2C2C2E',
		fontSize: 40,
		marginBottom: 24
	},
	facultyTitleText: {
		fontFamily: 'Lexend_300Light',
		fontSize: 16,
		marginBottom: 24
	},
	majorTitleText: {
		marginTop: 24,
		fontFamily: 'Lexend_300Light',
		fontSize: 16,
		marginBottom: 24
	},
	optionalTitleText: {
		marginTop: 24,
		fontFamily: 'Lexend_300Light',
		fontSize: 14
	},
	optionalText: {
		fontFamily: 'Lexend_300Light',
		fontSize: 14,
		marginBottom: 24
	},
	dropdownOne: {
		borderBottomWidth: 1,
		borderBottomColor: '#CBCBCB',
		marginBottom: 24
	},
	dropdownTwo: {
		borderBottomWidth: 1,
		borderBottomColor: '#CBCBCB',
		marginBottom: 24
	},
	dropdownPlaceholder: {
		color: '#CBCBCB'
	},
	loginPressable: {
		padding: 16,
		backgroundColor: '#EF7C00',
		borderRadius: 12
	},
	loginText: {
		textAlign: 'center',
		color: 'white',
		fontSize: 18,
		fontFamily: 'Lexend_400Regular'
	},
	toastContainer: {
		width: '90%',
		borderRadius: 8,
		paddingHorizontal: 10,
		paddingVertical: 8,
		flexDirection: 'row',
		alignItems: 'center',
		shadowColor: '#000',
		shadowOffset: { width: 0, height: 2 },
		shadowOpacity: 0.8,
		shadowRadius: 2,
		elevation: 5
	},
	warningToast: {
		backgroundColor: '#D00E17'
	},
	successToast: {
		backgroundColor: '#28a745'
	},
	textContainer: {
		flex: 1,
		flexShrink: 1
	},
	toastText1: {
		fontSize: 16,
		color: 'white'
	},
	toastText2: {
		fontSize: 12,
		color: 'white'
	},
	closeButton: {
		padding: 5
	},
	closeButtonText: {
		fontSize: 18,
		fontWeight: 'bold',
		color: 'white'
	}
});
