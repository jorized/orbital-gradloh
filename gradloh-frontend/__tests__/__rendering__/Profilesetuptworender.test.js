import React from 'react';
import TestRenderer, { act } from 'react-test-renderer';
import { NavigationContainer, useNavigation, useRoute } from '@react-navigation/native';
import ProfileSetUpTwoScreen from '../../screens/Auth/ProfileSetUpTwoScreen';
import { AuthContext } from '../../contexts/AuthContext';
import { AxiosContext } from '../../contexts/AxiosContext';


jest.mock('react-native-toast-message', () => {
  const React = require('react');
  const Toast = React.forwardRef((props, ref) => <div ref={ref} {...props} />);
  return {
    show: jest.fn(),
    hide: jest.fn(),
    default: Toast,
  };
});

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({ goBack: jest.fn() }),
  useRoute: () => ({
    params: {
      nickname: 'test',
      email: 'test@u.nus.edu',
      password: 'Password123%',
      enrolmentYear: '2022-2023'
    }
  }),
}));

const mockAuthContextValue = {
  authState: { authenticated: false },
  setAuthState: jest.fn(),
  getAccessToken: jest.fn(),
  logout: jest.fn(),
};

const mockAxiosContextValue = {
  publicAxios: {
    post: jest.fn().mockResolvedValue({ data: { message: 'Success' } }),
  },
  authAxios: {
    post: jest.fn().mockResolvedValue({ data: { message: 'Success' } }),
  },
};

describe('ProfileSetUpTwoScreen', () => {
  it('renders correctly', async () => {
    let tree;

    await act(async () => {
      tree = TestRenderer.create(
        <AuthContext.Provider value={mockAuthContextValue}>
          <AxiosContext.Provider value={mockAxiosContextValue}>
            <NavigationContainer>
              <ProfileSetUpTwoScreen />
            </NavigationContainer>
          </AxiosContext.Provider>
        </AuthContext.Provider>
      ).toJSON();
    });

    expect(tree).toMatchSnapshot();
  });
});
