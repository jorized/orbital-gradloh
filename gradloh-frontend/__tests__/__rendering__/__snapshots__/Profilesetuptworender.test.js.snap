// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileSetUpTwoScreen renders correctly 1`] = `null`;
